jsonpb/jsonpb.go

```diff
@@ -644,7 +649,19 @@ func (u *Unmarshaler) UnmarshalNext(dec *json.Decoder, pb proto.Message) error {
 // permutations of the related Marshaler.
 func (u *Unmarshaler) Unmarshal(r io.Reader, pb proto.Message) error {
 	dec := json.NewDecoder(r)
-	return u.UnmarshalNext(dec, pb)
+	if err := u.UnmarshalNext(dec, pb); err != nil {
+		return err
+	}
+
+	// Marshal it back to check for missing required field error.
+	// TODO: Parse through the message and check for any missing required field w/o having to
```

Comment:
This TODO will need to be resolved before this could be considered for merging. Marshaling the JSON just after Unmarshaling is going to build in a lot of overhead that wouldn’t be a good idea to pass around. (Especially for everyone using proto3, where required fields have been entirely removed.)

Reply:
Agree that this needs to be improved.  Sorry if I wasn't clear in my initial posting.  I didn't intend to merge this yet.  Since I didn't have time this week to continue doing this, I decided to post this PR for others to try it out and see if anyone may be interested in improving it further.

Note though that a proto3 message can contain a proto2 message and vice-versa.  So, as far as I know, the code will still need to traverse all the fields and subfields of the message looking for required ones that are not set.